internal/adapters/http/handler.go

```diff
@@ -0,0 +1,51 @@
+package http

```

Comment:
substituiria esse teu handler por algo como:


```suggestion
// internal/handler/order_handler.go
package handler

import (
	"ecommerce/internal/entity"
	"ecommerce/internal/usecase"
	"github.com/gin-gonic/gin"
	"net/http"
)

type OrderHandler struct {
	UseCase usecase.PlaceOrderUseCase
}

// Factory clean!
func NewOrderHandler(usecase usecase.PlaceOrderUseCase) *OrderHandler {
	return &OrderHandler{UseCase: usecase}
}

func (h *OrderHandler) PlaceOrder(c *gin.Context) {
	var order entity.Order
	if err := c.ShouldBindJSON(&order); err != nil {
		c.JSON(http.StatusBadRequest, gin.H{"mensagem": "anterior"}) // vc ja tem isso ai 
		return
	}
	if err := h.UseCase.Execute(&order); err != nil {
		c.JSON(http.StatusInternalServerError, gin.H{"mensagem": err.Error()}) // o mesmo aqui
		return
	}
	c.JSON(http.StatusOK, gin.H{"message": "order created"}) // ja tem a mensagem salva ai tbm
}
```